lms/djangoapps/discussion/tasks.py

```diff
@@ -90,3 +93,26 @@ def _get_thread_url(context):
         'id': context['thread_id'],
     }
     return urljoin(settings.LMS_ROOT_URL, permalink(thread_content))
+
+
+def _generate_ga_pixel_url(course_key, user_id):
+    # used for analytics
+    query_params = {
+        'v': '1',
+        't': 'event',
+        'ec': 'email',
+        'ea': 'open',
+        'tid': get_value("GOOGLE_ANALYTICS_TRACKING_ID", settings.GOOGLE_ANALYTICS_TRACKING_ID),
```

Comment:
Note that if this is being run by a celery worker then `get_value` might not do what you expect since there is no request available. We had to carefully pass the `site_id` down into all of our celery tasks so that we could check site-specific configuration.

Reply:
Related question for you @mulby - is `Site.objects.get_current()` not appropriate for obtaining a Site object to pass into `get_base_template_context()`?  We do this here: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/discussion/tasks.py#L79

Note that it's in the context of a celery task (and that a waffle switch is currently blocking this task from executing).

Reply:
Oh, wait, reading comprehension...ok, from your first comment, I think we need to do the same thing w.r.t. passing the `site_id` into our task.